### ISSUE TITLE
Invalid warning for string mistakenly assumed as array on read assignment

### ISSUE THREAD

**Issue by Rfferrao87** [koalaman/shellcheck#2073]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2178, SC2128
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

read -rp"(optional): " abc

abc="abc"

echo "${abc}"
```

#### Here's what shellcheck currently says:
```
Line 5	SC2178: Variable was used as an array but is now assigned a string.
Line 7	SC2128: Expanding an array without an index only gives the first element.
```
#### Here's what I wanted or expected to see:
```
No issues detected!
```

This appears to be happening when inserting a parentheses enclosed string with `read`s `-p` (_prompt_) parameter.

**Comment by koalaman** [comment]
You're right, this was a bug in the ad-hoc options parser, where `-rpa foo` would be parsed as `-r -p -a foo` instead of `-r -p a foo`, since it did not account for `-p` taking an argument. This has now been fixed.

In previous versions, you can work around it by adding a space: `read -rp "(optional): " abc`
